packages/backend/src/open_payments/payment/outgoing/service.ts

```diff
@@ -239,6 +243,11 @@ async function cancelOutgoingPayment(
   })
 }
 
+// TODO: Basically does all the logic in the validate spent amounts.
+// This felt more natural/easier to implement to me. Than how the task describes splitting
+// the paths in thre create function and only calling validate for the legacy path.
+// Not sure if it introduces any edges cases or jives as well with future issues (failed
+// payments for example).
 async function createOutgoingPayment(
```

Comment:
Note that this does not implement the logic quite how the issue specified. Had some issues when implementing that and found doing everything in the validate function to be much easier (otherwise I think we have to duplicate a bunch of logic there or chop it up into pieces for reuse - kind of a more tedious refactor either way).